lib/plugins/aws/deploy/lib/uploadArtifacts.test.js

```diff
@@ -286,33 +286,33 @@ describe('uploadArtifacts', () => {
 
       const uploadZipFileStub = sinon
         .stub(awsDeploy, 'uploadZipFile').resolves();
-      sinon.stub(fs, 'statSync').returns({ size: 1024 });
+      const statSyncStub = sinon.stub(fs, 'statSync').returns({ size: 1024 });
 
       return awsDeploy.uploadFunctions().then(() => {
         expect(uploadZipFileStub.calledTwice).to.be.equal(true);
         expect(uploadZipFileStub.args[0][0])
           .to.be.equal(awsDeploy.serverless.service.functions.first.package.artifact);
         expect(uploadZipFileStub.args[1][0])
           .to.be.equal(awsDeploy.serverless.service.package.artifact);
-        awsDeploy.uploadZipFile.restore();
-        fs.statSync.restore();
+        uploadZipFileStub.restore();
+        statSyncStub.restore();
       });
     });
 
     it('should log artifact size', () => {
       awsDeploy.serverless.config.servicePath = 'some/path';
       awsDeploy.serverless.service.service = 'new-service';
 
-      sinon.stub(fs, 'statSync').returns({ size: 1024 });
-      sinon.stub(awsDeploy, 'uploadZipFile').resolves();
+      const statSyncStub = sinon.stub(fs, 'statSync').returns({ size: 1024 });
+      const uploadZipFileStub = sinon.stub(awsDeploy, 'uploadZipFile').resolves();
```

Comment:
If you create the stubs inside the test, you have to make sure that they are restored if the test exists. Even if it enters a rejection path. Also use the `sandbox` to create the stubs. That makes sure that they are removed properly as soon as the sanbox is cleaned up.

So having this, you must add a `.finally()` clause to the test that restores and removes the stubs again. Otherwise following tests are affected.

Reply:
In general, to prevent such stub leaks, it's better to setup the test stubs in `before()` or `beforeEach()` and clean them in the resp. after methods.

Reply:
I can rewrite these tests to use the sandbox, but I don't think it related to this PR.
The changes I made were only to make sure these tests won't randomly fail [(1)](https://travis-ci.org/serverless/serverless/jobs/335682396) [(2)](https://travis-ci.org/serverless/serverless/jobs/335706593)

Reply:
Agree. It's out of scope. So the stabilization of the unit tests should include this